### PR TITLE
Fix comparing valueFrom in function-controller

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- as in title

Comparing string representation should be enough, see https://raw.githubusercontent.com/kubernetes/api/master/core/v1/generated.pb.go -> search for `func (this *EnvVarSource) String() string {`

**Related issue(s)**

### CODE REVIEW

File: components/function-controller/internal/controllers/serverless/function_controller.go [hunk 1]
```diff
@@ -48,7 +48,9 @@ func (r *FunctionReconciler) envsEqual(existing, expected []corev1.EnvVar) bool
 	for key, value := range existing {
 		expectedValue := expected[key]
 
-		if expectedValue.Name != value.Name || expectedValue.Value != value.Value || expectedValue.ValueFrom != value.ValueFrom { // valueFrom check is by reference
+
+
+		if expectedValue.Name != value.Name || expectedValue.Value != value.Value || expectedValue.ValueFrom.String() != value.ValueFrom.String() { // valueFrom check is by string representation
```
Comment (magicmatatjahu) [inline components/function-controller/internal/controllers/serverless/function_controller.go:53]: If think that before string comparison, we should check existing of pointers.

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
Please also add to `buildDeployment` method these lines:

```go
	envs := make([]corev1.EnvVar, 0)
	envs = append(envs, envVarsForDeployment...)
	envs = append(envs, rtmConfig.RuntimeEnvs...)
	envs = append(envs, instance.Spec.Env...)
```

Why? Fn-controller adds default envs at the end of array, and envs from user at the beginning, so user cannot redefine envs. See https://github.com/kyma-project/kyma/blob/master/components/function-controller/internal/controllers/serverless/build_resources.go#L331

File: components/function-controller/internal/controllers/serverless/function_controller.go (hunk 1 above)
Comment (aerfio) [reply components/function-controller/internal/controllers/serverless/function_controller.go:53]: it's already done by String() method

**Review by magicmatatjahu — DISMISSED** [review]

**Review by magicmatatjahu — DISMISSED** [review]

**Review by m00g3n — APPROVED** [review]

**Pull request merged**